### DISCUSSION TITLE
Best practise for customization/i8n?

### DISCUSSION THREAD

**Discussion by larsneo** [lucky-media/cloud#8] [Q&A] [answered]
As someone completly new to Statamic i'm really impressed by the various possibilities of the system. The setup was quite easy but now i'm thinking about the 'best practise' for customization and i8n. Is there any how-to for translation and especially update-safe customization of the theme (terms, fonts, colors, etc)?

**Comment by lokmanm** [comment, marked as answer]
Hey @larsneo, glad you asked that I will try to give you a thorough answer.

For multi language we recommend working with [Statamic Multi Site](https://statamic.dev/multi-site#content), this will basically `duplicate` your website in the required language and you can add content there in the language of your choosing.

As for the look and feel the website is configured with [Tailwind](https://tailwindcss.com/) which makes it very easy to do customizations. At the root of the project you will find a file called `tailwind.config.js`, there you can configure the colors or the font.

Before you do any customizations please make sure you run `npm install` so all the required dependencies are installed and then you can either run `npm run dev` for a one-time compilation, or you can do a `npm run watch` for continuous watching of the files.

If you want to install a new font you will need to head out to [FontSource](https://fontsource.org/fonts), choose a font there, install it with `npm` and then include it to the `app.js` file which is located in `resources/js/app.js`.

As for updates, the starter kits included with Statamic are a one-time installations, so there is no way to update it as it might break your website. You can read more about that [here](https://statamic.dev/installing-a-starter-kit#content). For an existing site I would suggest following the releases on this repo if we fix something although most of the components are battle tested. We regularly update the packages just to keep it updated to the latest versions.

Let me know if I answered your questions, or if you might have more.

Thanks

**Reply by larsneo** [reply]
Thx for your response :-) It's just a about a german only website so i just need to change e.g. datetime-format and some expressions like "read more". Based on my experience with the zikula application framework i was thinking about some kind of i8n under the hood - but hardcoding is ok anyway. Just wanted to make sure that i don't break my site with an update of the theme.

"As for updates, the starter kits included with Statamic are a one-time installations, so there is no way to update it as it might break your website." - didn't know about this and it leads me to another question: how can i make sure that the theme-related libraries (besides statamic core) are up2date (due to security concerns)?

**Reply by lokmanm** [reply]
For all those kind of stuff and internationalization I would suggest giving a read the Statamic docs as they are very thorough with stuff like this.

The only theme related libraries that we ship with these theme can be found on the [package.json](https://github.com/lucky-media/cloud/blob/main/package.json) file.

Most of our packages if not all are used only in the development process, so we actually just ship [alpine.js](https://alpinejs.dev/) as a dependency and its the only one you should update ( in case of a security concern ). 

We will regularly update packages so you can checkout the `package.json` file for updates in this repo in order to stay up to date.

If you want to be super safe, we use `alpine` only for the navigation, so it's easy to remove it as a dependency, you will just have to do the menu open/show functionality in plain javascript. If you decide to do this there will be no library left that can impose any security concern, as for the others are only development dependencies and they don't get shipped to the user.

I hope I was clear, let me know if you need anything else :)
